src/coreclr/jit/morph.cpp

```diff
@@ -13886,6 +13886,10 @@ void Compiler::fgMorphBlock(BasicBlock* block)
                             assertionsOut = pred->bbAssertionOutIfFalse;
                         }
                     }
+                    else
+                    {
+                        assertionsOut = pred->bbAssertionOut;
+                    }
```

Comment:
Small drive-by fix for some UB here (in the `useCondAssertions` case we were accessing an inactive union member).